graylog2-web-interface/src/components/visualizations/StackedGraphVisualization.jsx

```diff
@@ -31,20 +33,21 @@ const StackedGraphVisualization = React.createClass({
     return {};
   },
   componentDidMount() {
-    this.renderGraph();
-    this.dataPoints = this._formatData(this.props.data);
+    this.renderGraph(this.props);
+    this.dataPoints = this._formatData(this.props);
     this.drawData();
   },
   componentWillReceiveProps(nextProps) {
     if (deepEqual(this.props, nextProps)) {
       return;
     }
 
+    this._updateSeriesNames(nextProps);
+    this.dataPoints = this._formatData(nextProps);
     if (nextProps.height !== this.props.height || nextProps.width !== this.props.width) {
       this._resizeVisualization(nextProps.width, nextProps.height);
+      this.renderGraph(nextProps);
```

Comment:
Do we need to call `renderGraph()` here? It looks like it is updating the series names and resizing the visualization (which are the two things dependent on props as far as I can see). Both those properties are also being updated with custom methods, so I guess we can save a re-render on this case.

Reply:
What do you mean with "custom methods"? From my tests the graph is not properly resized without the following `renderGraph()` call.

Reply:
I mean that it looks like `renderGraph()`, `_updateSeriesNames()` and `_resizeVisualization()` are updating the same things, just at different times. If `_renderGraph()` is the one that is really needed to update the values, then I guess the other two methods are somehow redundant.